flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1082,6 +1110,13 @@ class JobManager(
 
         log.info(s"Using restart strategy $restartStrategy for $jobId.")
 
+        val jobMetrics = jobManagerMetricGroup match {
+          case Some(group) =>
+            group.addJob(jobGraph.getJobID, jobGraph.getName)
```

Comment:
What happens if `group.addJob` returns a `null` value? Then the instantiation of the `ExecutionGraph` would fail with a `NullPointerException`. We should check that and return an `UnregisteredMetricsGroup` instead.